controllers/servers/utils.py

```diff
@@ -601,8 +601,11 @@ def generate_addons_replication_request(request, replication_type):
     volume_id_info = get_volume_id_info(request.volume_id)
     volume_internal_id = volume_id_info.ids.internal_id
 
-    other_volume_id_info = get_volume_id_info(request.replication_id)
-    other_volume_internal_id = other_volume_id_info.ids.internal_id
+    if replication_type == REPLICATION_TYPE_MIRROR:
+        other_volume_id_info = get_volume_id_info(request.replication_id)
+        other_volume_internal_id = other_volume_id_info.ids.internal_id
+    else:
+        other_volume_internal_id = None
```

Comment:
I think it can be exported to `_get_other_volume_internal_id` function

Reply:
Done